File: QuanLyKhachSanApp/QuanLyKhachSanApp/Models/DatDichVu.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace QuanLyKhachSanApp.Models
{
    [Table("DatDichVu")]
    public class DatDichVu
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Required]
        public int DatDichVuID { get; set; }

        [Required]
        public int SoLuong { get; set; }

        public int? DichVuID { get; set; }

        public int? HoaDonID { get; set; }

        [ForeignKey("DichVuID")]
        public virtual DichVu DichVu { get; set; }

        [ForeignKey("HoaDonID")]
        public virtual HoaDon HoaDon { get; set; }

    }
}

File: QuanLyKhachSanCli/src/models/DatDichVu.ts
import { DichVu } from "@/models/DichVu";
import { HoaDon } from "@/models/HoaDon";

export interface DatDichVu {
    DatDichVuID: number;
    SoLuong: number;
    DichVuID: number;
    HoaDonID: number;
    DichVu?: DichVu;
    HoaDon?: HoaDon;
}
